### ISSUE TITLE
Expired downloads, 401 error on /api/info/

### ISSUE THREAD

**Issue by gaycookie** [timvisee/send#76]
Dear,

I'm not running it as a Docker container but I do run it behind a Nginx reverse proxy.
When I click on Download or open the url I get an Download Expired message.
And when I open the Developer Tools in Chrome I notice an 401 error on `/api/info/...`

I'm running it via npm directly on my machine using: `npm run prod`.
And using the below config on my Nginx.

```
server {
  server_name <domain>;
  client_max_body_size 2560M;

  location / {
    proxy_pass http://localhost:8080;
    proxy_http_version 1.1;

    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
  }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/<domain>/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/<domain>/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
}
server {
  if ($host = <domain>) {
    return 301 https://$host$request_uri;
  }

  listen 80;
  server_name <domain>;
  return 404;
}
```

**Comment by timvisee** [comment]
Are you seeing any errors in the node log?

When running in production, usage of Redis is required. Errors like this may pop up if a connection Redis fails. I believe usage of S3 is required in production as well, though I'm not 100% about that at the moment.

**Comment by gaycookie** [comment]
> Are you seeing any errors in the node log?

No, I've not seen any error so far that was actually why I went here.

> When running in production, usage of Redis is required.

I did not know about this, I just setup a Redis server quickly and everything works fine now!
Is this explained somewhere? Because I guess I missed that part 😊

> I believe usage of S3 is required in production as well,

So far now it I run it with Redis together, I do not seem to need it for the functionality I need.
I'm able to upload, download and all without setup anything on S3 (I do not even use Amazon services 😉)

I wanna thank you a lot and I consider my problem solved!

**Comment by timvisee** [comment]
> Is this explained somewhere?

The requirements in the README list it, but I agree that it isn't super clear: https://github.com/timvisee/send#requirements

> I wanna thank you a lot and I consider my problem solved!

Awesome! Happy to help.

**Comment by gaycookie** [comment]
> The requirements in the README list it, but I agree that it isn't super clear: https://github.com/timvisee/send#requirements

I think I missed the part it said "for development" and just thought it was optional 😊
